lib/src/idg/seq.rs

```diff
@@ -0,0 +1,180 @@
+use crate::err::Error;
+use crate::kvs::{Key, Val};
+use roaring::RoaringTreemap;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+// Id is a unique monotonic id generated by the sequence generator.
+pub(crate) type Id = u64;
+
+// Seq is a sequence generator that can be used to generate unique monotonic ids.
+// It can reuse freed ids by keeping track of them in a roaring bitmap.
+// It is used to generate ids for any SurrealDB objects that need aliases (e.g. namespaces, databases, tables, indexes, etc.)
+#[derive(Clone)]
+pub struct Seq {
+	state_key: Key,
+	available_ids: Option<RoaringTreemap>,
+	next_id: Id,
```

Comment:
If we don't need strict monotonicity of IDs we might partition the next ID ranges by SurrealDB nodes.

That way, each node can give unique IDs out of it's ID range without reading and writing back the Seq state in every transaction.

It will look conceptually similar to how a TiDB AUTO_INCREMENT variant works as far as I remember :)

But that's another story. I'd like to start with this strictly-monotonic and simple implementation.